swift-sdk/IterableAPIInternal.swift

```diff
@@ -616,10 +616,11 @@ import Foundation
             let notificationMetadata = IterableNotificationMetadata.metadata(fromInAppOptions: messageId)
             
             DispatchQueue.main.async {
-                IterableInAppManager.showIterableNotificationHTML(html, trackParams: notificationMetadata, callbackBlock: callbackBlock, backgroundAlpha: backgroundAlpha, padding: edgeInsets)
+                let opened = IterableInAppManager.showIterableNotificationHTML(html, trackParams: notificationMetadata, callbackBlock: callbackBlock, backgroundAlpha: backgroundAlpha, padding: edgeInsets)
+                if opened {
+                    self.inAppConsume(messageId)
+                }
             }
-
-            self.inAppConsume(messageId)
```

Comment:
The ObjC implementation also consumes if 'href' isn't found (otherwise the invalid in-app may stay at the top of the queue until expiration), line 609 here.

Reply:
Not sure I follow.

Reply:
If you look at the check at line 608, it prints an error, but still tries to display the invalid in-app message (it shouldn't) and consumes it (it should). After logging the error, it should do `self.inAppConsume(messageId)` and `return` if the message is invalid.